src/Utils/removeUnreachable.ts

```diff
@@ -12,9 +14,9 @@ function addReachable(
     }
 
     if (definition.$ref) {
-        const typeName = decodeURIComponent(definition.$ref.slice(14));
-        if (reachable.has(typeName)) {
-            // we've already processed this definition
+        const typeName = decodeURIComponent(definition.$ref.slice(DEFINITION_OFFSET));
+        if (reachable.has(typeName) || !isLocalRef(definition.$ref)) {
```

Comment:
I think this is too late. At this point we already generated the schema. It would be better to not even recurse into types when we have a ref.

Reply:
Good point. I'll look into it.

Reply:
I updated `AnnotatedNodeParser` such that it short-circuits child type parsing if an explicit `ref` annotation was provided.
Instead, it sets the child type to `AnyType`. The AnyType is conveniently represented as an empty object which means we'll only get the `$ref` property in the output schema and no additional noise coming from the type itself.